test/utils/resource.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
```

Comment:
I suggested wrapping `cli` in an object:

```
type K8sCRUD struct {
  cli ctrlClient.Client
}

func (c *K8sCRUD) Create(ctx context.Context, ...) { ... }
func (c *K8sCRUD) Delete(ctx context.Context, ...) { ... }
...
```
```

Reply:
Created test/utils/crud.go for it.